src/DocumentFragment.js

```diff
@@ -0,0 +1,35 @@
+  // ### DocumentFragment Shim
+  //
+  // A representation of a document fragment that can be normally generated
+  // by calling `document.createDocumentFragment`. Since the document fragment
+  // is itself a node, it inherits all properties of a node.
+module.exports = (function(){
+  "use strict";
+
+  var Node = require("./Node");
+
+  function DocumentFragment() {
+    this.node = new Node(Node.DOCUMENT_FRAGMENT_NODE, "#document-fragment");
+    this.node._ownerDocument = this;
```

Comment:
this really should be `null`, not `this`. A node should never be its own parent, that's going to introduce so many bugs in the future when doing `element.parentNode` evaluation that it's best we prevent that ever being a problem right now